### PR TITLE
Add ssdt i2c-xosi device (list WIP)

### PR DESCRIPTION
**Pull request by Lorys89**

### CODE REVIEW

**Comment by kprinssu** [comment]
Thanks for the PR @Lorys89.  Can you please provide more context as to why you want this patch added to repository? Can you also update the VoodooI2C documentation as to how one can use the SSDT you provided. 

Right now as it stands, the ACPI code will declare `OSYS` object and it will not be referenced anywhere else. I suspect you want people to use a modified XOSI patch but I do not see that in the SSDT.

**Comment by m0d16l14n1** [comment]
> Thanks for the PR @Lorys89. Can you please provide more context as to why you want this patch added to repository? Can you also update the VoodooI2C documentation as to how one can use the SSDT you provided.
> 
> Right now as it stands, the ACPI code will declare `OSYS` object and it will not be referenced anywhere else. I suspect you want people to use a modified XOSI patch but I do not see that in the SSDT.

I have a problem with XOSI patch > glitching sound and fan, but i need it to get touchpad working, so by injecting OSYS value inside touchpad scope i got touchpad working and my sound and fan are good. So that's much better for using in some cases. 

Usually XOSI patch + SSDT recommended for laptop users (ex. Dortania guides), but sometimes it bring some problems, so i think it will be good to have example of SSDT patch for users who have problems with XOSI method.

**Comment by kprinssu** [comment]
Thanks for the clarification @m0d16l14n1. 

@Lorys89 The wording for the documentation is a bit rough and has few grammtical errors.  Can you let me know when you have something ready to be merged? I'll take a closer look when you are done with the PR.

Tagging @ben9923 so he can take a look as well.

**Comment by Lorys89** [comment]
> Thanks for the clarification @m0d16l14n1.
> 
> @Lorys89 The wording for the documentation is a bit rough and has few grammtical errors. Can you let me know when you have something ready to be merged? I'll take a closer look when you are done with the PR.
> 
> Tagging @ben9923 so he can take a look as well.

I'm sorry, I'm working from cell at the moment, as soon as I get to the pc, I take care of the documentation and the revision of grammatical errors.

**Comment by Lorys89** [comment]
by default I put the value windows 2012(win 8) 0x07DC, if you prefer windows 2015 (win 10) I modify ssdt, in most of the acpi analyzed under i2c I found 0x07DC, in rare cases 0x07DD (win 8.1)

File: Documentation/Installation.md [hunk 1]
```diff
@@ -53,7 +53,8 @@ You **must** be familiar with DSDT patching. If you do not know how to patch you
 
 You must be using the latest version of MaciASL found [here](https://bitbucket.org/RehabMan/os-x-maciasl-patchmatic/downloads/) (or an equivalent ACPI patcher). You must ensure that MaciASL's compiler version is set to the latest one. This can be done in the `iASL` tab of MaciASL's Preferences. After applying each patch, save your DSDT and restart your system.
 
-You can try SSDT hotpatch method, you need [SSDT-EC](https://github.com/acidanthera/OpenCorePkg/blob/master/Docs/AcpiSamples/Source/SSDT-EC.dsl) with SSDT-I2C-XOSI, many i2c devices activate using this method.
+You can try the basic method, which consists of using SSDT patches, to start you need "SSDT-EC" and "SSDT-I2C-XOSI", now you have to integrate them into the ACPI folder of your EFI and declare in your config.plist, after integrating any changes, save your config.plist and reboot the system.
```
Comment (kprinssu) [inline Documentation/Installation.md:56]: How about:

```
VoodooI2C has a "quick method" to get the hardware to work, which consists of using SSDT patches and ACPI hot-patching. To start you need `SSDT-EC.dsl`[LINK TO DORTANIA DSL FILE] and `SSDT-I2C-XOSI.dsl`[MARKDOWN_LINK_TO_SSDT_DSL]. You will also need to let your bootloader load the compiled AML files and apply the OSI to XOSI patch for your I2C device.
```

**Comment by kprinssu** [comment]
> by default I put the value windows 2012(win 8) 0x07DC, if you prefer windows 2015 (win 10) I modify ssdt, in most of the acpi analyzed under i2c I found 0x07DC, in rare cases 0x07DD (win 8.1)

@ben9923 what's your suggestion on this, as this very hardware and ACPI centric.

**Review by Lorys89 — COMMENTED** [review]
ok like this?

**Comment by ben9923** [comment]
In my opinion, instead of providing a generic SSDT the guide should instruct in which cases it's needed (Not all devices depend on `OSYS`) and how to create one specific for the user's device.
It already instructs users to find the correct ACPI path for their device. Windows 10 also sounds like a safe option to me.

Anyway, I'm not sure how EC is related (Mentioned in the `Installation.md` changes).

**Comment by Lorys89** [comment]
> In my opinion, instead of providing a generic SSDT the guide should instruct in which cases it's needed (Not all devices depend on `OSYS`, though most indeed reference it in `_INI` and `_CRS` - something to check) and how to create one specific for the user's device.
> It already instructs users to find the correct ACPI path for their device. Windows 10 also sounds like a safe option to me.
> 
> While instructing to do something like that, we may also instruct to patch the GPIO control variable that's often mentioned in the same `_INI` and `_CRS` methods.
> That way we have a full hotpatching-compatible guide.
> 
> Anyway, I'm not sure how EC is related (Mentioned in the `Installation.md` changes).

[photo.zip](https://github.com/VoodooI2C/VoodooI2C/files/5972882/photo.zip)
You mean a guide that explains how to locate the acpi path of your i2c device under windows, and then switch to macos to create specific ssdt, ok? so a complete guide?

see the photos in the archive.

**Comment by ben9923** [comment]
> > In my opinion, instead of providing a generic SSDT the guide should instruct in which cases it's needed (Not all devices depend on `OSYS`, though most indeed reference it in `_INI` and `_CRS` - something to check) and how to create one specific for the user's device.
> > It already instructs users to find the correct ACPI path for their device. Windows 10 also sounds like a safe option to me.
> > While instructing to do something like that, we may also instruct to patch the GPIO control variable that's often mentioned in the same `_INI` and `_CRS` methods.
> > That way we have a full hotpatching-compatible guide.
> > Anyway, I'm not sure how EC is related (Mentioned in the `Installation.md` changes).
> 
> [photo.zip](https://github.com/VoodooI2C/VoodooI2C/files/5972882/photo.zip)
> You mean a guide that explains how to locate the acpi path of your i2c device under windows, and then switch to macos to create specific ssdt, ok? so a complete guide?
> 
> see the photos in the archive.
> I hope you liked my basic idea :)

Yeah. The current guide already covers that, just for static patching. It needs to be updated for hotpatching and making minimal adjustments only (i.e. setting control variables rather than swapping a method).

**Comment by Lorys89** [comment]
i close the pr, to draft hotpatching guide.
See you soon😘

**Pull request closed without merging**